crates/red_knot_workspace/tests/check.rs

```diff
@@ -272,7 +272,7 @@ const KNOWN_FAILURES: &[(&str, bool, bool)] = &[
     ("crates/ruff_linter/resources/test/fixtures/pyupgrade/UP039.py", true, false),
     // related to circular references in type aliases (salsa cycle panic):
     ("crates/ruff_python_parser/resources/inline/err/type_alias_invalid_value_expr.py", true, true),
-    // related to f-strings in annotations (invalid syntax)
+    // related to circular references in f-string annotations (invalid syntax)
```

Comment:
The examples below look similar to the new corpus test added here. They also need the fix in this PR. But they require an *additional* fix, because they are weirdly self-referential and contain something like
```py
x: f"{x}"
```